src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb

```diff
@@ -1272,6 +1282,7 @@ lVbRuntimePlus:
             Return New VisualBasicCommandLineArguments With
             {
                 .IsScriptRunner = IsScriptRunner,
+                .InteractiveMode = InteractiveMode Or (IsScriptRunner AndAlso sourceFiles.Count = 0),
```

Comment:
This line determines whether `vbi` will run as a REPL or will execute a script. It considers the following cases:
1. `vbi` starts a REPL
2. `vbi file.vb` runs the script without starting a REPL
3. `vbi file.vb -i` starts a REPL since `-i` option was explicitly specified